sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -182,6 +190,48 @@ case class GlobalLimitExec(limit: Int, child: SparkPlan) extends BaseLimitExec {
     copy(child = newChild)
 }
 
+/**
+ * Skip the first `offset` elements then take the first `limit` of the following elements in
+ * the child's single output partition.
```

Comment:
Can there be multiple output partitions in the case that the child was not sorted?

In the case it wasn't sorted, we can return an arbitrary collection of rows of the correct size, so it doesn't really matter, just trying to understand the invariants.

Reply:
It looks like if the child has multiple partitions, zipWithIndex will index starting with all the rows in the first partition, then the next partition etc. This is fine assuming the child data isn't sorted.